### PR TITLE
1.0.0

### PR DESCRIPTION
**Pull request by zoontek**
- Remove `url` from location, add `toString`
- Add `raw` object in `location` for easier comparaisons
- Use `useSyncExternalStoreWithSelector` for `useRoute` and `useRoutes`
- Fix keyboard focus reset on search params / hash only update
- Multiple methods renaming
-  Add a basic Link component

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -1,7 +1,19 @@
 export const first = <T>(value: T[]): T | undefined => value[0];
+export const identity = <T>(value: T): T => value;
 export const isNonEmpty = (value: string): boolean => value !== "";
 export const isParam = (value: string): boolean => value.startsWith(":");
 
+export const areParamsArrayEqual = (arrayA: string[], arrayB: string[]) =>
+  !(
+    arrayA.length !== arrayB.length ||
+    arrayA.findIndex((value, index) => value !== arrayB[index]) !== -1
+  );
```
Comment (bloodyowl) [inline src/helpers.ts:10]:
```suggestion
export const areParamsArrayEqual = (arrayA: string[], arrayB: string[]) =>
  arrayA.length === arrayB.length &&
  arrayA.every((a, index) => a === arrayB[index]);
```

File: src/index.ts [hunk 2]
```diff
@@ -137,17 +138,27 @@ export const createRouter = <
...
+        };
+      },
+      undefined,
+      identity,
+      areItemKeysEqual,
+    );
 
-    return JSON.parse(routes);
+    // @ts-expect-error
+    return routes;
```
Comment (bloodyowl) [inline src/index.ts:161]: In the end, I feel like we can remove that entire function (as we found another more efficient way to handle breadcrumbs patterns internally)

File: src/helpers.ts (hunk 1 above)
Comment (zoontek) [reply src/helpers.ts]: @bloodyowl I shouldn't code on saturday morning

File: src/index.ts (hunk 2 above)
Comment (zoontek) [reply src/index.ts]: Good for me. I never really liked it.

**Pull request merged**